modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/LakeWriter.scala

```diff
@@ -128,14 +129,23 @@ object LakeWriter {
   /**
    * Implementation of the LakeWriter
    *
-   * The mutex is needed because we allow overlapping windows. They prevent two different windows
+   * The mutexes are needed because we allow overlapping windows. They prevent two different windows
    * from trying to run the same expensive operation at the same time.
```

Comment:
If I understand correctly, the risk is that the second window wants to start doing the same operation while the first one is not over yet, right?

Is it "only" because the operations are expensive that we want only one at a time or also for other reasons?

Reply:
For `mutexForRemoteWriting`, it is only because the operation is expensive.  For `mutexForLocalAppending` it is nothing to do with expensive operations, it is only to prevent a race condition.

This is my mistake, I forgot to delete two lines, which has made everything more confusing.